quickwit-indexing/src/models/indexed_split.rs

```diff
@@ -72,31 +92,34 @@ impl IndexedSplit {
         let box_mmap_directory = Box::new(mmap_directory);
         let controlled_directory =
             ControlledDirectory::new(box_mmap_directory, progress, kill_switch);
-        let index = index_builder.open_or_create(controlled_directory.clone())?;
-        let index_writer = index.writer_with_num_threads(
-            1, // DO NOT MODIFY THIS!
-            // This is not something that we want to use in quickwit.
-            indexing_resources.heap_size.get_bytes() as usize,
-        )?;
-        let split_attrs = SplitAttrs {
-            split_id,
-            partition_id,
-            pipeline_id,
-            num_docs: 0,
-            uncompressed_docs_size_in_bytes: 0,
-            time_range: None,
-            replaced_split_ids: Vec::new(),
-        };
-        index_writer.set_merge_policy(Box::new(NoMergePolicy));
-        Ok(IndexedSplit {
-            split_attrs,
-            index,
+        let index_writer =
+            index_builder.single_segment_index_writer(controlled_directory.clone(), 10_000_000)?;
```

Comment:
Shouldn't we use `indexing_resources.heap_size.get_bytes()` to set the memory budget?